app/controllers/application_controller.rb

```diff
@@ -51,4 +52,15 @@ def complete_request(action: nil, path: nil)
   def authorize!
     raise NoMethodError
   end
+
+  def check_college
```

Comment:
Ok, below set `@current_college` to `College.first || College.new` and add a flash message if it isn't `persisted`.